jailer/src/lib.rs

```diff
@@ -97,6 +96,122 @@ pub enum Error {
+            CloseNetNsFd(ref err) => write!(f, "Failed to close netns fd: {}", err),
+            CloseDevNullFd(ref err) => write!(f, "Failed to close /dev/null fd: {}", err),
+            Copy(ref file, ref path, ref err) => write!(
+                f,
+                "Failed to copy {:?} to {:?}: error {:?}",
+                file, path, err
+            ),
+            CreateDir(ref path, ref err) => {
+                write!(f, "Failed to create directory {:?}: error {:?}", path, err)
+            }
+            CStringParsing(_) => write!(f, "Encountered interior \\0 while parsing a string"),
+            Dup2(ref err) => write!(f, "Failed to duplicate fd: {}", err),
+            Exec(ref err) => write!(f, "Failed to exec into Firecracker: error {:?}", err),
+            FileName(ref path) => write!(f, "Failed to extract filename from path {:?}", path),
+            FileOpen(ref path, ref err) => {
+                write!(f, "Failed to open file {:?}: error {:?}", path, err)
+            }
+            FromBytesWithNul(ref bytes) => {
+                write!(f, "Failed to decode string from byte array: {:?}", bytes)
+            }
+            GetOldFdFlags(ref err) => write!(f, "Failed to get flags from fd: {}", err),
+            Gid(ref gid) => write!(f, "Invalid gid: {}", gid),
+            InvalidInstanceId(ref err) => write!(f, "Invalid instance ID: {}", err),
+            MissingArgument(ref arg) => write!(f, "Missing argument: {}", arg),
+            MissingParent(ref path) => write!(f, "File {:?} doesn't have a parent", path),
+            MkdirOldRoot(ref err) => write!(
+                f,
+                "Failed to create the jail root directory before pivoting root: {}",
+                err
+            ),
+            MknodDevNetTun(ref err) => write!(
+                f,
+                "Failed to create /dev/net/tun via mknod inside the jail: {}",
+                err
+            ),
+            MountBind(ref err) => {
+                write!(f, "Failed to bind mount the jail root directory: {}", err)
+            }
+            MountPropagationPrivate(ref err) => write!(
+                f,
+                "Failed to change the propagation type to private: {}",
+                err
+            ),
+            NotAFile(ref path) => write!(f, "{:?} is not a file", path),
+            NumaNode(ref node) => write!(f, "Invalid numa node: {}", node),
+            OpenDevKvm(ref err) => write!(f, "Failed to open /dev/kvm: {}", err),
+            OpenDevNull(ref err) => write!(f, "Failed to open /dev/null: {}", err),
+            OsStringParsing(ref path, _) => {
+                write!(f, "Failed to parse path {:?} into an OsString", path)
+            }
+            PivotRoot(ref err) => write!(f, "Failed to pivot root: {}", err),
+            ReadLine(ref path, ref err) => {
+                write!(f, "Failed to read line from {:?}: error {:?}", path, err)
```

Comment:
```suggestion
                write!(f, "Failed to read line from {:?}: {}", path, err)
```

Reply:
The inner error is an `io::Error`: `ReadLine(PathBuf, io::Error)` - which doesn't implement `Display`, so formatting with `{}` won't work.